src/test/perl/path.t

```diff
@@ -290,45 +292,103 @@ rmtree if -d $basetest;
 ok($mc->any_exists($basetest), "any_exists true on created directory");
 ok(! $mc->file_exists($basetest), "file_exists false on created directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on created directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on created file");
 ok($mc->any_exists($basetestfile), "any_exists true on created file");
 ok($mc->file_exists($basetestfile), "file_exists true on created file");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on created file");
+
+# noreset=1
+verify_exception("existence tests do not reset exception/fail", "origfailure", 0, 1);
+ok($mc->_reset_exception_fail(), "_reset_exception_fail after existence tests");
 
-# Test (broken) symlink and _exsists methods
 
-ok(symlink("really_really_missing", $brokenlink), "broken symlink created");
-makefile("$basetest/tgtdir/tgtfile");
-ok(symlink("tgtdir", $dirlink), "directory symlink created");
-ok(symlink("tgtdir/tgtfile", $filelink), "file symlink created");
+=head2 symlink/hardlink creation/update/test
+
+=cut
+
+# Test symlink creation
+
+my $target_directory = "tgtdir";
+my $target_file1 = "tgtfile1";
+my $target_file2 = "$basetest/$target_directory/tgtfile2";
+makefile("$basetest/$target_file1");
+makefile($target_file2);
+my %opts;
+is($mc->symlink($target_directory, $dirlink), CHANGED, "directory symlink created");
+is($mc->symlink($target_file2, $filelink), CHANGED, "file symlink created");
+is($mc->symlink($target_file2, $filelink), SUCCESS, "file symlink already exists: nothing done");
+is($mc->symlink($target_file1, $filelink), CHANGED, "file symlink already exists: nothing done");
+my $link_status = $mc->symlink($target_file1, $target_file2);
+ok(! defined($link_status), "symlink failed: existing file not replaced by a symlink");
+is($mc->{fail},
+   "Failed to create symlink $target_file2 (target=$target_file1)",
+   "fail attribute set after symlink failure (existing file not replaced by a symlink)");
+$opts{force} = 1;
+is($mc->symlink($target_file1, $target_file2, %opts), CHANGED, "existing file replaced by a symlink (force option set)");
+$link_status = $mc->symlink($target_file1, "$basetest/$target_directory", %opts);
+ok (! defined($link_status), "directory not replaced by a symlink (force option set)");
+is($mc->{fail},
+   "Failed to create symlink $basetest/$target_directory (target=$target_file1)",
+   "fail attribute set after symlink failure (existing directory not replaced by a symlink)");
+$opts{check} = 1;
+ok(! $mc->symlink("really_really_missing", $brokenlink, %opts), "broken symlink not created (target existence enforced)");
+delete $opts{nocheck};
+is($mc->symlink("really_missing", $brokenlink), CHANGED, "broken symlink created");
+
+# Test hardlink creation
+my $cwd = cwd();
+my $hardlink1 = "$basetest/$target_directory/hardlink1";
+my $hardlink2 = "$basetest/$target_directory/hardlink2";
+# hardlink target must be an absolute path or the link path directory is prepended
```

Comment:
this "feature" should be documented in the pod. it's certainly new to me. and you should add a test for this behaviour; these details.